org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Parser.java

```diff
@@ -1398,34 +1384,29 @@ protected ParameterizedQualifiedTypeReference computeQualifiedGenericsFromRightS
 		tokensSize ++;
 	} else if (rightSide instanceof SingleTypeReference) {
 		tokensSize ++;
-	} else if (rightSide instanceof ParameterizedQualifiedTypeReference) {
-		tokensSize += ((QualifiedTypeReference) rightSide).tokens.length;
```

Comment:
Why is this removed ? I doesn't appear any of the above `instanceof` checks would render it useless.

Reply:
The case of `ParameterizedQualifiedTypeReference` is already handled in the very exact same way in the next `instanceof QualifiedTypeReference`, so those like are actually useless. But as they're not related to instanceof, I can keep them and create another commit for it if you think it makes things better or easier.